### PR TITLE
Move from defunct codeowners to GitHub CODEOWNERS.

### LINKED ISSUE #3755: Switch from defunct `codeowners` directory to new GH codeowners file
https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

Did not exist when I first set up the code owners thing. There was an action which originally was supposed to auto-assign reviewers, but it broke pretty quickly and so the codeowners dir is defunct.

### PR DESCRIPTION
**Pull request by dtzSiFive**
Migrated some, self-assigned a bunch, expect this to be tweaked.

cc @teqdruid @darthscsi @seldridge @fabianschuiki -- please take a look as I've volunteered you a bit (or migrated from old setup).

Fixes #3755.

### CODE REVIEW

**Comment by dtzSiFive** [comment]
This should be tweaked, but have to step away, LMK folks how you want to proceed!

**Comment by seldridge** [comment]
Nice. I'm good with switching (and I don't think we were really using the other anyway?). Thanks for taking the initiative.

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+**/FIRRTL* @darthscsi @dtzSiFive @seldridge
+*/firtool @dtzSiFive
+libs/Firtool @dtzSiFive
+
+# LLHD
+**/LLHD* @fabianschuiki
+tools/llhd-sim @fabianschuiki
+
+# Handshake
+**/Handshake* @hanchenye @stephenneuendorffer
```
Comment (mikeurbach) [inline CODEOWNERS:20]: This should probably be @mortbopet @Dinistro @RamirezLucas now

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+# ESI
+**/Dialect/ESI @teqdruid
+
+# FIRRTL
+**/FIRRTL* @darthscsi @dtzSiFive @seldridge
+*/firtool @dtzSiFive
+libs/Firtool @dtzSiFive
+
+# LLHD
```
Comment (mikeurbach) [inline CODEOWNERS:15]: Maybe throw @maerhart under LLHD and Arc?

**Review by mikeurbach — COMMENTED** [review]
I think it makes sense to do this the usual way with CODEOWNERS, and I think this has been discussed before. So +1 from me to move this direction.

I assume this is just a first pass to move to the new format from the existing ownership?

I have some suggestions for some of the newer dialects / conversions, but that can be another PR.

File: CODEOWNERS (hunk 2 above)
Comment (maerhart) [reply CODEOWNERS:15]: Sure, that's fine by me.

File: CODEOWNERS (hunk 1 above)
Comment (dtzSiFive) [reply CODEOWNERS:20]: Great, thanks. Done!

File: CODEOWNERS (hunk 2 above)
Comment (dtzSiFive) [reply CODEOWNERS:15]: Yes, updated!

**Comment by dtzSiFive** [comment]
> I don't think we were really using the other anyway?

I don't think so, and the linked issue says that it broke quickly / is defunct so I think the old bit was just dead for some time now.

> I assume this is just a first pass to move to the new format from the existing ownership?

Yes, that's right! I'd rather not get things entirely wrong to start of course but yep mostly let's get to new format and adjust as needed (even immediately via PR's as you say).  Also we'll see what sorts of patterns make most sense as things continue to grow/evolve.

----

There are options to require approval but for now let's just get some sense of ownership and maybe facilitate who is asked for review when!

**Comment by dtzSiFive** [comment]
Alright, I'm going to go ahead and merge -- please adjust (or revert!) as needed if there are objections or problems, but this should be a good starting point.

I'm not sure where between "Ownership" and "subscribe to changes to these files" we want this to be, but we'll sort that out :smile:.

**Pull request merged**